onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1923,6 +1923,248 @@ Matrix product that behaves like numpy.matmul: https://docs.scipy.org/doc/numpy-
         FusedMatMulShapeInference(ctx);
       });
 
+    static const char* NonZero_doc = R"DOC(
+  Behaves similar to https://github.com/onnx/onnx/blob/master/docs/Operators.md#NonZero
+  returns a tensor that contains coordinates for non-zero entries. Even though the input
+  is a sparse tensor, that usually does not contain zeros, the model logic may construct
+  sparse tensor containing explicit zeros that convey model specific meaning.
+  If the input sparse tensor has 2-D COO indices, the output shape would be (nnz,2) with 2 coordinates per
+  non-zero entry. If the input sparse tensor contains 1-D COO indices, the output would contain flat indices
+  of shape (nnz).
+
+  XXX: Should be make the output shape fixed regardless on the input COO indices shape? 
+)DOC";
```

Comment:
Are you thinking of (nnz*2) & (nnz) instead of (nnz,2) & (nnz) respectively?
Our interest is to use `Gather` to gather these non zero indices from a Sparse Tensor - essentially remove zero values and return the resultant Sparse Tensor. Specifically need it for rank-1 right now. Implementing rank-2 "Filter" functionality this way is more complex for rank-2.